pkg/controller/noderefutil/providerid_test.go

```diff
@@ -17,32 +17,62 @@ limitations under the License.
 package noderefutil
 
 import (
-	"strings"
 	"testing"
-
-	"github.com/pkg/errors"
 )
 
 func TestNewProviderID(t *testing.T) {
-	input1 := "aws:////instance-id"
-	_, err := NewProviderID(input1)
-	if err != nil {
-		t.Fatalf("Expected no errors, got %v", err)
+	tests := []struct {
+		name string
+		input string
+		expectedID string
+	} {
+		{
+			name: "one segment",
+			input: "aws://instance-id",
+			expectedID: "instance-id",
+		},
+		{
+			name: "multiple segments",
+			input: "aws://one/two/three/instance-id",
```

Comment:
How about a test case with empty segments as well? Such as `aws:////instance-id`?

Reply:
Sure, we should reject that, right?

Reply:
Added it to the errors test

Reply:
No, it should be accepted, it is one of the allowed values by the cloud provider

Reply:
ok, will fix

Reply:
For the common case back from aws specifically, we should expect to see something similar to `aws:///us-east-1a/instance-id`, but `aws:////instance-id` is also acceptable.